tests/ga/test_monitor.py

```diff
@@ -832,26 +842,33 @@ def test_send_extension_metrics_telemetry_with_actual_cgroup(self, patch_add_eve
     def test_report_event_metrics_sent_for_actual_cgroup(self, patch_report_event, http_get, patch_get_protocol, *args):
         self.assertTrue(i_am_root(), "Test does not run when non-root")
 
-        max_num_polls = 5
-        time_to_wait = 1
-        extn_name = "foobar-1.0.0"
-
-        cgs = make_new_cgroup(extn_name)
-        self.assertEqual(len(cgs), 2)
-
-        ext_handler_properties = ExtHandlerProperties()
-        ext_handler_properties.version = "1.0.0"
-        ext_handler = ExtHandler(name='foobar')
-        ext_handler.properties = ext_handler_properties
-        ext_handler_instance = ExtHandlerInstance(ext_handler=ext_handler, protocol=None)
-        ext_handler_instance.set_operation("Enable")
-
-        monitor_handler = get_monitor_handler()
-        monitor_handler.init_protocols()
-        monitor_handler.last_cgroup_polling_telemetry = datetime.datetime.utcnow() - timedelta(hours=1)
-        monitor_handler.last_cgroup_report_telemetry = datetime.datetime.utcnow() - timedelta(hours=1)
-
-        command = self.create_script("keep_cpu_busy_and_consume_memory_for_{0}_seconds".format(time_to_wait), '''
+        # This test has some timing issues when systemd is managing cgroups, so we force the file system API
```

Comment:
how did all those timing issues surfaced?

Reply:
Initially, the systemd way of creating cgroup would not create a new cgroup when testing and I was seeing memory data (0s) and no CPU data (we throw an exception), which was messing up the data collection and the intended purpose of this test. Fixing it to FileSystem API helped me to run a 3-second running test and helped the data for the test.

Reply:
OK. Collection of cgroup data is always done using the file system, even under systemd. Instead of adding this hack to each of these tests you may simple execute them explicitly using FileSystemCgroupsApi. See, for example, the tests for the cgroup configurator.